substrate/frame/staking-async/src/tests/unbonding_queue.rs

```diff
@@ -0,0 +1,603 @@
+				back_of_unbonding_queue: Zero::zero(),
+			})
+		));
+
+		Session::roll_until_active_era(2);
+		Session::roll_until_active_era(3);
+		Session::roll_until_active_era(4);
+		assert_eq!(current_era(), 4);
+		// There are no stakers
+		assert_eq!(EraLowestRatioTotalStake::<Test>::get().into_inner(), vec![0, 0, 0]);
+		assert_eq!(Staking::get_min_lowest_stake(), 0);
+
+		// Create validators with different stakes for the next era.
+		let exposures = to_bounded_supports(vec![
+			(1, Support { total: 1000, voters: vec![] }),
+			(2, Support { total: 2000, voters: vec![] }),
+			(3, Support { total: 3000, voters: vec![] }),
+			(4, Support { total: 4000, voters: vec![] }),
+		]);
+
+		// Trigger new era to calculate the lowest proportion.
+		assert_ok!(EraElectionPlanner::<T>::do_elect_paged_inner(
+			exposures.clone().try_into().unwrap()
+		));
+		Session::roll_until_active_era(5);
+		assert_eq!(EraLowestRatioTotalStake::<Test>::get().into_inner(), vec![0, 0, 1000]);
+		// The lowest proportion is 33% of 4 validators ~ 1.32.
+		// Hence, the lowest 1 validator with 1000.
+		assert_eq!(Staking::get_min_lowest_stake(), 0);
+
+		assert_ok!(EraElectionPlanner::<Test>::do_elect_paged_inner(exposures.clone()));
+		Session::roll_until_active_era(6);
+		assert_eq!(EraLowestRatioTotalStake::<Test>::get().into_inner(), vec![0, 1000, 1000]);
+		assert_eq!(Staking::get_min_lowest_stake(), 0);
+
+		// Ensure old entry is pruned after bonding duration (3 eras).
+		assert_ok!(EraElectionPlanner::<Test>::do_elect_paged_inner(exposures.clone()));
+		Session::roll_until_active_era(7);
+		assert_eq!(EraLowestRatioTotalStake::<Test>::get().into_inner(), vec![1000, 1000, 1000]);
+		assert_eq!(Staking::get_min_lowest_stake(), 1000);
+	});
+}
+
+#[test]
+fn get_unbond_eras_delta_with_zero_max_unstake_works() {
+	ExtBuilder::default().build_and_execute(|| {
+		let config = UnbondingQueueConfig {
+			min_slashable_share: Perbill::from_percent(50),
+			lowest_ratio: Perbill::from_percent(34),
+			unbond_period_lower_bound: 1,
+			back_of_unbonding_queue: 0,
+		};
+
+		// Must be the maximum because the minimum lowest stake is zero, which defaults to
+		// the maximum unbonding period.
+		assert_eq!(EraLowestRatioTotalStake::<Test>::get().into_inner(), vec![]);
+		assert_eq!(Staking::get_unbonding_delta(1, config), 3000000000000);
+
+		let config = UnbondingQueueConfig {
+			min_slashable_share: Perbill::from_percent(0),
+			lowest_ratio: Perbill::from_percent(34),
+			unbond_period_lower_bound: 1,
+			back_of_unbonding_queue: 0,
+		};
+
+		// Now the minimum lowest stake is also zero because the slashable share is zero, so
+		// again it must be the maximum unbonding period.
+		assert_ok!(EraLowestRatioTotalStake::<Test>::try_append(1000));
+		assert_eq!(Staking::get_unbonding_delta(1, config), normalize_era(3));
+	});
+}
+
+#[test]
+fn get_unbond_eras_delta_works() {
+	ExtBuilder::default().build_and_execute(|| {
+		let config = UnbondingQueueConfig {
```

Comment:
Again, I urgue you to: 

1. Remove `back_of_unbonding_queue` from the `UnbondingQueueConfig`, it is not a configuration but rather a state variable. 
2. `UnbondingQueueConfig` should be more clear in what values should be set such that this feature is a noop
3. A helper function `ExtBuilder` should enable it for said tests.

Reply:
First part refactored in [1866f07](https://github.com/paritytech/polkadot-sdk/pull/8298/commits/1866f076aa40cc6001a19fdad2b83be4a74ce359).

Reply:
Second part done in [88449ef](https://github.com/paritytech/polkadot-sdk/pull/8298/commits/88449ef4718d8c8867bf7b0501ed528e52fdb797).

Reply:
Actually I took the liberty of doing it that way in [3d05285](https://github.com/paritytech/polkadot-sdk/pull/8298/commits/3d052850f1266521b012d5f8e30f6c04bdf0d81d), as it makes the tests cleaner.